java-security/README.md

```diff
@@ -55,42 +65,96 @@ In case of XSUAA does the JWT provide a valid `jku` token header parameter that
 
-Alternatively, you can also specify your own Service Configuration:
+### `OAuth2ServiceServiceConfiguration` usage
+Access the OAuth2 Service configuration bound to the application:
+* For Xsuaa Service
+  ```java  
+  OAuth2ServiceConfiguration serviceConfig = Environments.getCurrent().getXsuaaConfiguration();
+  OAuth2ServiceConfiguration serviceConfig = Environments.getCurrent().getXsuaaConfigurationForTokenExchange(); // in case of multiple xsuaa bindings, this method returns the broker plan binding
+  ```
+* For Identity service
+  ```java
+  OAuth2ServiceConfiguration serviceConfig = Environments.getCurrent().getIasConfiguration(); 
+  ```
+:bulb: `Environments` auto-detects the environment(Cloud Foundry or Kubernetes) the app is deployed in. 
+
+Alternatively, you can also specify a custom Service Configuration:
+ ```java
+ OAuth2ServiceConfiguration serviceConfig = OAuth2ServiceConfigurationBuilder.forService(Service.XSUAA)
+         .withProperty(CFConstants.XSUAA.APP_ID, "appid")
+         .withProperty(CFConstants.XSUAA.UAA_DOMAIN, "authentication.sap.hana.ondemand.com")
+         .withUrl("https://paas.authentication.sap.hana.ondemand.com")
+         .withClientId("oauth-client")
+         .withClientSecret("oauth-client-secret")
+        .build();
+ ```
+
+Use custom Service Configuration in `XsuaaTokenAuthenticator` or `IasTokenAuthenticator` as shown below:
 ```java
-OAuth2ServiceConfiguration serviceConfig = OAuth2ServiceConfigurationBuilder.forService(Service.XSUAA)
-      .withProperty(CFConstants.XSUAA.APP_ID, "appid")
-      .withProperty(CFConstants.XSUAA.UAA_DOMAIN, "authentication.sap.hana.ondemand.com")
-      .withUrl("https://paas.authentication.sap.hana.ondemand.com")
-      .withClientId("oauth-client")
-      .withClientSecret("oauth-client-secret")
-      .build();
+XsuaaTokenAuthenticator authenticator = new XsuaaTokenAuthenticator()
+						.withServiceConfiguration(OAuth2ServiceConfiguration);
 ```
-:bulb: `OAuth2ServiceConfiguration.getClientIdentity()` is a convenience method that with `OAuth2ServiceConfigurationBuilder` implementation will resolve `ClientCredentials` or `ClientCertificate` implementations of `ClientIdentity` interface based on the `credential-type` from the Xsuaa service binding. Therefore, providing the correct implementation for the configured authentication type e.g. X.509 or client secret based.
 
+:bulb: `OAuth2ServiceConfiguration.getClientIdentity()` offers a convenience method that with `OAuth2ServiceConfigurationBuilder` 
+implementation will resolve `ClientCredentials` or `ClientCertificate` implementations of `ClientIdentity` interface based on the `credential-type` from the Xsuaa or Identity service binding
+and with that providing the correct implementation for the configured authentication type e.g. X.509 or client secret based.
+ 
 #### :mega: Service configuration in Kubernetes/Kyma environment 
```

Comment:
I wouldn't remove it because it's still relevant for k8s users to know how the binding secrets needs to be provided for the lib to work and that for example service-operator is only supported out of the box by SBA

Reply:
Okay, I restored it in spring-security as well.